### PR TITLE
[SUBMARINE-200] Submarine project manager ProjectRestApi

### PR DESCRIPTION
**Pull request by LinhaoZhu**
### What is this PR for?
1. org.apache.submarine.database.service.ProjectService
2. ProjectMapper.xml, ProjectFilesMapper.xml
3. org.apache.submarine.rest.ProjectRestApi
4. database table: project, project_files;

### What type of PR is it?
[ Feature ]

### Todos
* [ ] - Task

### What is the Jira issue?
* [SUBMARINE-200](https://issues.apache.org/jira/browse/SUBMARINE-200)

### How should this be tested?
* [CI Pass](https://travis-ci.org/LinhaoZhu/hadoop-submarine/builds/590285482)

### Questions:
* Does the licenses files need update? Yes/No
* Is there breaking changes for older versions? Yes/No
* Does this needs documentation? Yes/No

### CODE REVIEW

File: submarine-workbench/submarine-server/src/main/java/org/apache/submarine/database/service/ProjectService.java [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    try (SqlSession sqlSession = MyBatisUtil.getSqlSession()) {
+      ProjectMapper projectMapper = sqlSession.getMapper(ProjectMapper.class);
+      projectMapper.updateByPrimaryKeySelective(project);
+
+      ProjectFilesMapper projectFilesMapper = sqlSession.getMapper(ProjectFilesMapper.class);
+      Map<String, Object> where = new HashMap<>();
+      where.put("projectId", project.getId());
+      // Take two lists of difference
+      List<ProjectFiles> old_projectFiles = projectFilesMapper.selectAll(where);
+      List<String> old_projectFiles_id = new ArrayList<>();
```
Comment (liuxunorg) [inline submarine-workbench/submarine-server/src/main/java/org/apache/submarine/database/service/ProjectService.java]: change `old_projectFiles_id ` to `oldProjectFilesId `

File: submarine-workbench/submarine-server/src/main/java/org/apache/submarine/database/service/ProjectService.java [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+          curr.setProjectId(project.getId());
+          projectFilesMapper.insert(curr);
+        }
+      }
+
+      sqlSession.commit();
+    } catch (Exception e) {
+      LOG.error(e.getMessage(), e);
+      throw new Exception(e);
+    }
```
Comment (liuxunorg) [inline submarine-workbench/submarine-server/src/main/java/org/apache/submarine/database/service/ProjectService.java]: need add 
```
 finally {
      sqlSession.close();
    }
```
Other places are also need add.

File: submarine-workbench/submarine-server/src/test/java/org/apache/submarine/database/service/ProjectServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+    project.setProjectFilesList(list);
+
+    Boolean ret = projectService.add(project);
+    assertTrue(ret);
+
+    List<Project> projectList = projectService.queryPageList("ProjectServiceTest-UserName",
+        "create_time", "desc", 0, 100);
+    assertEquals(projectList.size(), 1);
+
+    Project project_db = projectList.get(0);
```
Comment (liuxunorg) [inline submarine-workbench/submarine-server/src/test/java/org/apache/submarine/database/service/ProjectServiceTest.java]: change `project_db` to `projectDb`

**Comment by liuxunorg** [comment]
@LinhaoZhu , Don't do multiple things in one PR.

File: submarine-workbench/submarine-server/src/main/java/org/apache/submarine/database/service/ProjectService.java (hunk 1 above)
Comment (LinhaoZhu) [reply submarine-workbench/submarine-server/src/main/java/org/apache/submarine/database/service/ProjectService.java]: Done.

File: submarine-workbench/submarine-server/src/main/java/org/apache/submarine/database/service/ProjectService.java (hunk 2 above)
Comment (LinhaoZhu) [reply submarine-workbench/submarine-server/src/main/java/org/apache/submarine/database/service/ProjectService.java]: https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html
Based on the description in the above article, I don't think it is necessary to actively close the Session.

File: submarine-workbench/submarine-server/src/test/java/org/apache/submarine/database/service/ProjectServiceTest.java (hunk 3 above)
Comment (LinhaoZhu) [reply submarine-workbench/submarine-server/src/test/java/org/apache/submarine/database/service/ProjectServiceTest.java]: Done.

**Comment by LinhaoZhu** [comment]
> @LinhaoZhu , Don't do multiple things in one PR.

I have deleted the irrelevant submissions.

**Comment by LinhaoZhu** [comment]
[CI Pass](https://travis-ci.org/LinhaoZhu/hadoop-submarine/builds/590285482)

**Comment by liuxunorg** [comment]
@LinhaoZhu , Thank you contributor.
LGTM

**Pull request closed without merging**